ec2/instance.py

```diff
@@ -622,3 +636,68 @@ def _get_instance_subnet(provider_variables):
             'instance may only be attached to one subnet')
 
     return list_of_subnets[0] if list_of_subnets else None
+
+
+def _get_network_interface_object(ec2_client, eni_id):
+    '''Gets a NetworkInterface object from AWS'''
+    eni_objs = ec2_client.get_all_network_interfaces(
+        network_interface_ids=[eni_id])
+    if not isinstance(eni_objs, list) or len(eni_objs) < 1:
+        raise NonRecoverableError(
+            'Could not get NetworkInterface object of {0}'
+            .format(eni_id))
+    return eni_objs[0]
+
+
+def _find_attached_eni():
+    '''
+        Finds all attached Network Interfaces via relationships
+
+        returns: Relationship context
+    '''
+    return [
+        rel for rel in ctx.instance.relationships
+        if NETWORK_INTERFACE_RELATIONSHIP in rel.type_hierarchy and
+        NETWORK_INTERFACE_GROUP_TYPE in rel.target.node.type_hierarchy
+    ]
+
+
+def _build_network_interfaces():
+    '''
+        Searches for attached ElasticNetworkInterface instances
+        and converts them into a NetworkInterfaceCollection for
+        use with run_instances()
+
+        :rtype: boto.ec2.networkinterface.NetworkInterfaceCollection
+        returns: EC2-compatible list of Elastic Network Interfaces
+    '''
+    network_interfaces = []
+    for rel in _find_attached_eni():
+        rt_props = rel.target.instance.runtime_properties
+        network_interfaces.append(NetworkInterfaceSpecification(
+            network_interface_id=rt_props.get('aws_resource_id'),
+            # subnet_id=rt_props.get('aws_subnet_id'),
+            # description=rt_props.get('aws_description'),
+            # private_ip_address=rt_props.get('aws_private_ip_address'),
+            # groups=rt_props.get('aws_security_groups', list()),
+            device_index=rt_props.get('aws_device_index'),
+            delete_on_termination=rt_props.get('aws_delete_on_termination')))
+    return NetworkInterfaceCollection(*network_interfaces)
+
+
+def _sanitize_json_input(us_data):
```

Comment:
does this have any useful application outside of ec2 instances that we can apply?